transactor/src/main/scala/io/mediachain/types/CborSerialization.scala

```diff
@@ -1,51 +1,123 @@
-package io.mediachain.transactor
+package io.mediachain.types
 
 import io.mediachain.multihash.MultiHash
 
 
 object CborSerialization {
 
   import cats.data.Xor
-  import io.mediachain.transactor.Dummies.DummyReference
-  import io.mediachain.transactor.Types._
+  import io.mediachain.types.Datastore._
   import io.mediachain.util.cbor.CborAST._
   import io.mediachain.util.cbor.CborCodec
 
   import scala.util.Try
 
+  /**
+    * A mapping of `MediachainType` to the `CborDeserializer` to use when
+    * decoding objects of that type.
+    *
+    * Used to allow the transactors to deserialize chain cells into the most
+    * generic representation, while allowing other peers to deserialize into
+    * specific subtypes.
+    *
+    * Defaults to deserializing into specific subtypes; to override this
+    * behavior and deserialize into generic `EntityChainCell`s and
+    * `ArtefactChainCell`s, make an implicit `DeserializerMap` in the scope
+    * where you call the various `fromCbor` methods.
+    *
+    * e.g:
+    * ```
+    * implicit val deserializerMap = CborSerialization.transactorDeserializers
+    *
+    * val fooCell = fromCborBytes(fooCellBytes)
+    * ```
+    */
   type DeserializerMap = Map[MediachainType, CborDeserializer[CborSerializable]]
```

Comment:
what if we took the witness approach to this... this way the passing around of `deserializers` wouldn't be so fhity

```scala
  trait Deserializer[T] {
    def deserialize(bytes: Array[Byte]): Option[T]
  }
  
  // dumb string deserialization
  implicit val deserializeInt = new Deserializer[Int] {
    override def deserialize(bytes: Array[Byte]): Option[Int] = {
      val str = new String(bytes)
      try {
        str.toInt
      } catch {
        case _ => None
      }
    }
  }

  def foo(myInt: Array[Byte])(implicit d: Deserializer[Int]): Int = {
    d.deserialize(myInt).map(_ + 1).getOrElse(0)
  }

  def generalDeserializer[T](implicit d: Deserializer[T]): Option[T] = {
     val bytes = fetchDb()
     d.deserialize(bytes)
  }
```